crates/ursa-pod/src/client.rs

```diff
@@ -77,24 +82,31 @@ where
                     ..
                 }) => {
                     // Receive proof
-                    let len = proof_len as usize;
-                    self.conn.read_buffer(len);
-                    let mut proof_buf = BytesMut::with_capacity(len);
-                    loop {
-                        match instrument!(
-                            self.conn.read_frame(None).await?,
-                            "tag=read_proof_buffer"
-                        ) {
-                            Some(UrsaFrame::Buffer(bytes)) => {
-                                debug!("recv proof chunk");
-                                proof_buf.put_slice(&bytes);
-                                if proof_buf.len() == len {
-                                    // todo: decode proof
-                                    break;
+                    if proof_len != 0 {
+                        let len = proof_len as usize;
+                        self.conn.read_buffer(len);
+                        let mut proof_buf = BytesMut::with_capacity(len);
+                        loop {
+                            match instrument!(
+                                self.conn.read_frame(None).await?,
```

Comment:
isn't read_frame supposed to return the frame when the data len species by `read_buffer(len)` is provided? is this loop necessary?

Reply:
I think this makes sense, I can refactor it